### ISSUE TITLE
Todo for V11

### ISSUE THREAD

**Issue by Yenthe666** [Yenthe666/auto_backup#72]
- Remove length limit on fields
- Add HTTPS support (new field HTTP/HTTPS?)
- Fix translation https://github.com/Yenthe666/auto_backup/blob/11.0/auto_backup/models/db_backup.py#L132 (add _() to https://github.com/Yenthe666/auto_backup/blob/11.0/auto_backup/models/db_backup.py#L119 , https://github.com/Yenthe666/auto_backup/blob/11.0/auto_backup/models/db_backup.py#L125 and https://github.com/Yenthe666/auto_backup/blob/11.0/auto_backup/models/db_backup.py#L126)
- Change the "Failed" to a boolean in memory at https://github.com/Yenthe666/auto_backup/blob/11.0/auto_backup/models/db_backup.py#L131 - it'll otherwise fail and a boolean is cleaner.

**Comment by Yenthe666** [comment]
First range of improvements & fixes at https://github.com/Yenthe666/auto_backup/commit/d6cf2a729580b3e42106d80d89f54cf0deb83e94

**Comment by jcchoquet** [comment]
Hi,
I have installed your module in V11, but the backup don't work everytime and i have not message in log.

**Comment by MartinHeisig** [comment]
Hi. while testing your version for Odoo 11 and getting errors with (like it seems) odoo backup and my used postgres version I found some minor errors in your code which would be nice to be fixed in a future version:

all in file 'db_backup.py'

missing

> import socket

causes
```
File "..../auto_backup/models/db_backup.py", line 30, in execute
    except socket.error as error:
NameError: name 'socket' is not defined
```

and if you solve this you will get
```
File "..../auto_backup.git/auto_backup/models/db_backup.py", line 33, in execute
    raise e
NameError: name 'e' is not defined
```
because 

> raise e

 should be 

> raise error

I hope this will info will help you.
After fixing this I will still hang on my error
```
File "..../odoo/tools/misc.py", line 128, in exec_pg_command
    raise Exception('Postgres subprocess %s error %s' % (args2, rc))
Exception: Postgres subprocess ('/usr/bin/pg_dump', '--no-owner', '--file=/tmp/tmppbm1t0ix/dump.sql', 'myDB') error 1
```
which seems to be a problem between Odoo11 and my used Postgresql version 10.1. It also raises while using backup over the database manger

**Comment by Yenthe666** [comment]
Hi @MartinHeisig,

Thanks a lot for the feedback and information. I've corrected both issues in https://github.com/Yenthe666/auto_backup/commit/90cc040df7cade0f56b3b9d6499f5514ac39a17f 
As for your third error: it looks like your Odoo databases aren't configured fine. I can't help you with this one I'm afraid.

**Comment by jcchoquet** [comment]
Hi, I have found my problem but I don't know how to solve.
My Odoo V11 is installed inside a docker, and at the first connection the admin_password (tools.config['admin_passwd']) is in clear but not the second time.
Can you help me ?

**Comment by JKE-be** [comment]
tools.config['admin_passwd'] is now hashed for security. 
So even if you set it in clear, at the first use that will be crypted

**Comment by jcchoquet** [comment]
ok, but for this line : bkp = execute(conn, 'dump', tools.config['admin_passwd'], rec.name, rec.backup_type) in file 'db_backup.py', if the password is not in clear, the command fails.

**Comment by MartinHeisig** [comment]
its funny. while searching for my error (which really seems to be a problem with postgresql10, because postgresql9.6 which comes with odoo11 works) I came along this issue too.
For testing purposes I set my password in this line in clear text and all works fine else not.

and thx for the fast fix.

**Comment by Yenthe666** [comment]
Hmm, I believe this is because Olivier Dony (from Odoo) changed the storing of the admin password. I believe he hashed it away instead of plain text. @JKE-be does this ring any bells to you? I can't find the related commit right away. I guess I should hash/encode the password then?

**Comment by JKE-be** [comment]
https://github.com/odoo/odoo/commit/7d16769

**Comment by Yenthe666** [comment]
Yup - looks like I need to add the `odoo.tools.config.verify_admin_password(passwd)` too in my module. @jcchoquet and @MartinHeisig feel free to try it already. 😉

**Comment by jcchoquet** [comment]
odoo.tools.config.verify_admin_password(passwd) return True or False, not the password... ;-)

**Comment by MartinHeisig** [comment]
As of now I have no clue for the passwort problem. I looked about but don't really understand because it shows different behaviour. First call after server start it gives the clear text password and evry try after the hashed version? And in the end they need the clear text password for the dumping function? That really confuses me...

Btw for my issue mentioned before - I solved it. I know its not a problem with this module but only a short explanation what was the cause. I have a separate server with a postgres10 and odoo uses this from an other server. The problem is/was that odoo forces a local postgres installation while installing. So my Odoo server had a local, but unused, postgres9.6. The backup function from odoo uses these local files in /usr/share/postgresql-common to trigger the dump on the database server. And because these files was from version 9.6 they wasnt able to communicate correctly with the version 10 server. solution was to update the unused local postgres to version 10 too so these files got updated and he was able to interact with the database server. logical but tricky to find.

**Comment by jcchoquet** [comment]
@Yenthe666, have you find a solution for the hashed password?

**Comment by Yenthe666** [comment]
@jcchoquet I haven't gotten around to it yet. Sorry!

**Comment by jcchoquet** [comment]
no problem... :-)

**Comment by gevasi** [comment]
I had this module running on my odoo 10 installation without problems. About a month ago I installed a fresh odoo 11 on ubuntu and installed this module and it works, but only once.
I have to restart the odoo server each time to make it run again.
Running the cron job manually also only works once, second time it doesn't do anything.
All i'm getting in the logs is this the first time:
2017-12-07 11:00:50,510 76224 INFO None werkzeug: 127.0.0.1 - - [07/Dec/2017 11:00:50] "POST /xmlrpc/db HTTP/1.1" 200 -
2017-12-07 11:00:50,543 76224 INFO None odoo.service.db: DUMP DB: gevasi format zip
2017-12-07 11:00:52,049 76224 INFO ? werkzeug: 127.0.0.1 - - [07/Dec/2017 11:00:52] "GET /web/static/src/img/spin.png HTTP/1.0" 200 -
2017-12-07 11:00:52,373 76224 INFO None werkzeug: 127.0.0.1 - - [07/Dec/2017 11:00:52] "POST /xmlrpc/db HTTP/1.1" 200 -
2017-12-07 11:00:52,628 76224 INFO ? paramiko.transport: Connected (version 2.0, client OpenSSH_6.8p1-hpn14v6)
2017-12-07 11:00:52,928 76224 INFO ? paramiko.transport: Authentication (password) successful!
2017-12-07 11:00:53,227 76224 INFO gevasi paramiko.transport.sftp: [chan 0] Opened sftp connection (server version 3)
2017-12-07 11:00:53,380 76224 INFO gevasi odoo.addons.auto_backup.models.db_backup: Copying File /opt/odoo/backups/07_12_2017_11_00_50_gevasi.zip------ success
2017-12-07 11:00:53,444 76224 INFO gevasi werkzeug: 127.0.0.1 - - [07/Dec/2017 11:00:53] "POST /web/dataset/call_button HTTP/1.0" 200 -
2017-12-07 11:00:53,508 76224 INFO gevasi werkzeug: 127.0.0.1 - - [07/Dec/2017 11:00:53] "POST /web/dataset/call_kw/ir.cron/read HTTP/1.0" 200 -

When I click the run manual button the second time I get this:
2017-12-07 11:02:27,653 76224 INFO None werkzeug: 127.0.0.1 - - [07/Dec/2017 11:02:27] "POST /xmlrpc/db HTTP/1.1" 200 -
2017-12-07 11:02:27,686 76224 INFO None werkzeug: 127.0.0.1 - - [07/Dec/2017 11:02:27] "POST /xmlrpc/db HTTP/1.1" 200 -
2017-12-07 11:02:27,687 76224 INFO gevasi werkzeug: 127.0.0.1 - - [07/Dec/2017 11:02:27] "POST /web/dataset/call_button HTTP/1.0" 200 -
2017-12-07 11:02:27,742 76224 INFO gevasi werkzeug: 127.0.0.1 - - [07/Dec/2017 11:02:27] "POST /web/dataset/call_kw/ir.cron/read HTTP/1.0" 200 -
Not getting any errors and the email fetching cron job is running normally so it's not cron either.
Anyone know why this is happening?
